### ISSUE TITLE
Add `media-feature-name-value-no-unknown`

### ISSUE THREAD

**Issue by mattxwang** [stylelint/stylelint#6560]
### What is the problem you're trying to solve?

In the discussion for https://github.com/stylelint/stylelint/pull/6550, @romainmenke mentioned certain types of issues involving invalid media query ranges. Their examples include:

```css
@media (width <= 100) {} /* number is not a length */
@media (width <= 50%) {} /* percentage is not a length */
@media (min-aspect-ratio: 16px/9px) {} /* length values do not make up a ratio */
```

I could also see the potential for auditing queries with the wrong number or types or arguments, ex:

```css
@media (width <= hover) /* invalid: hover is a keyword, not a length */
@media (color <= 16px)  /* invalid: color expects an integer */

/* grid is *not* a range, but just to also call out: */
@media (grid: 10)   /* invalid: grid expects a boolean (<mq-boolean>) of 1 or 0 */
@media (grid: 10px) /* invalid: grid expects a boolean (<mq-boolean>) of 1 or 0 */
```

I'm not sure if this is an appropriate rule for Stylelint, especially given its potential complexity.

### What solution would you like to see?

- Name: `media-feature-name-value-no-unknown`
- Primary option: `true`
- Secondary options: none
- Autofixable: No
- Message: `Unexpected unknown value "${value}" for media feature "${name}"`
- Description: Disallow unknown values for media features.
- Section: "Avoid Errors" -> "Unknown"
- Extended description: something about valid value types for specific media features

**Comment by ybiquitous** [comment]
@mattxwang Thanks for opening the issue!

**Comment by romainmenke** [comment]
It would be relatively straightforward with our media query parser to check if a media query is syntactically valid or not.

Types of errors :
1. the `@media` params or not even remotely similar to a media query
   - `isMediaQueryInvalid(x)` will be true
   - e.g. `@media @foo {}`
2. the type is invalid
   - `isMediaQueryWithType(x)` will be true
   - `x.getMediaType()` is not one of `all | print | screen | tty | tv | projection | handheld | braille | embossed | aural | speech`
   - e.g. `@media tablet and (min-width: 800px)`
3. the parenthesis and combinators are incorrect
   - `isGeneralEnclosed(x)` will be true
   - e.g. `@media ((min-width: 300px) and (hover: hover) or (aspect-ratio: 1 / 1))`
4. syntactically incorrect values for media queries in general are used
   - `isGeneralEnclosed(x)` will be true
   - e.g. `@media (min-width: var(--foo))`
   - e.g. `@media (min-width: 50%)`
5. incorrect values are used for a specific media feature
   - `isMediaFeature(x)` will be true
   - requires manual validation of values and units against : https://drafts.csswg.org/mediaqueries-5/#media-descriptor-table
   - e.g. `@media (width: coarse)`

I think `4.` and `5.` are good fits for a `media-feature-name-value-no-unknown` rule.

`1.`, `2.` and `3.` will likely require one or more other rules.

**Comment by ybiquitous** [comment]
@romainmenke Thanks for the reminder. Your suggestion sounds great to me. 👍🏼 

Could you please redraw this new rule's blueprint?

**Comment by romainmenke** [comment]
I've made some edits to the blueprints

**Comment by ybiquitous** [comment]
Thank you!

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new rule](https://stylelint.io/developer-guide/rules#add-a-rule) in the Developer guide.

**Comment by romainmenke** [comment]
I've got a first working draft of the new rule ready here : https://github.com/stylelint/stylelint/pull/6906#issuecomment-1579104068

I don't expect anyone to review this at this time, but if someone is curious or has early feedback, that is always welcome 🙇 

I tried to focus only on value validation and I am ignoring all other types of errors.

Some other types of errors I thought of while writing this rule:
- `(--custom-mq: 100px)` : invalid because custom media must be in boolean features
- `(min-width)` : invalid because `min-` prefix can not be used in boolean features
- ...

Technically these are also errors related to values and names but I think these are more related to the general grammar. i.e. wrong location of a value or missing value.

Only when the following is true does the rule kick in:
- the media query has valid syntax
- the media feature has valid syntax
- there is a media feature name
- there is a media feature value
